src/core/meshcop/dataset_manager_ftd.cpp

```diff
@@ -63,75 +63,146 @@ ActiveDataset::ActiveDataset(ThreadNetif &aThreadNetif):
     mResourceSet(OPENTHREAD_URI_ACTIVE_SET, &ActiveDataset::HandleSet, this)
 {
     mCoapServer.AddResource(mResourceGet);
+    memset(mActiveDatasetTlvs, 0, sizeof(mActiveDatasetTlvs));
+}
+
+bool ActiveDataset::isTlvInitialized(Tlv::Type aType, uint8_t aLength)
+{
+    bool rval = false;
+
+    VerifyOrExit(aLength > 0,);
+
+    for (uint8_t i = 0; i < aLength; i++)
+    {
+        if (mActiveDatasetTlvs[i] == aType)
```

Comment:
Do we need to have a separate array for this?  Can we just iterate over the TLVs directly and avoid the extra memory overhead?